role_appearance/src/Theme/ThemeNegotiator.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Drupal\role_appearance\Theme;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\role\Service\RoleControlManager;
+use Drupal\Core\Theme\ThemeNegotiatorInterface;
+
+/**
+ * Role Theme Negotiator.
+ */
+class ThemeNegotiator implements ThemeNegotiatorInterface {
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $user;
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Drupal\role\Service\RoleControlManager definition.
+   *
+   * @var \Drupal\role\Service\RoleControlManager
+   */
+  protected $roleControlManager;
+
+  /**
+   * Creates a new AdminNegotiator instance.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   The current user.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\role\Service\RoleControlManager $role_manager
+   *   The role manager.
+   */
+  public function __construct(AccountInterface $user, ConfigFactoryInterface $config_factory, RoleControlManager $role_manager) {
+    $this->user = $user;
+    $this->configFactory = $config_factory;
+    $this->roleControlManager = $role_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function determineActiveTheme(RouteMatchInterface $route_match) {
+    $custom_theme = $this->roleControlManager->getRoleThirdPartySetting($this->user, 'role_theme');
```

Comment:
`role_theme` it is module name? If yes, it should be `const` into your service

Reply:
no this is plugin id